library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java

```diff
@@ -55,11 +64,45 @@ public PojoInnerJoin(int num, Class<?> outClass, String... keys)
 
     this.keys = Arrays.copyOf(keys, keys.length);
     this.outClass = outClass;
+    firstTimeStream = true;
+    firstTimeStream2 = true;
+    createSetters();
+  }
+
+  private void createSetters()
+  {
+    Field[] fields = outClass.getDeclaredFields();
+    setters = new ArrayList<>();
+    for (Field field : fields) {
+      Class outputField = ClassUtils.primitiveToWrapper(field.getType());
+      String fieldName = field.getName();
+      setters.add(new KeyValPair<>(fieldName,PojoUtils.createSetter(outClass,fieldName,outputField)));
+    }
+  }
+
+  private void createGetters(Class<?> input, int streamIndex)
+  {
+    Field[] fields = input.getDeclaredFields();
+    List<KeyValPair<String,PojoUtils.Getter>> getters = new ArrayList<>();
+    for (Field field : fields) {
+      Class inputField = ClassUtils.primitiveToWrapper(field.getType());
+      String fieldName = field.getName();
+      getters.add(new KeyValPair<>(fieldName,PojoUtils.createGetter(input, fieldName, inputField)));
+    }
+    if (streamIndex == 1) {
+      gettersStream1 = getters;
+    } else {
+      gettersStream2 = getters;
+    }
```

Comment:
you could return the getters and assign them later. instread assign to correct variables using streamIndex. This pattern looks very common in multiple operator and creating getters and setters for multiple expressions could be made as part of PojoUtils or PojoUtilsHelpers.